### DISCUSSION TITLE
Monthly gem5 Developers' Meeting - October 10th at 4pm (UTC)

### DISCUSSION THREAD

**Discussion by ivanaamit** [gem5/gem5#1607] [Announcements]
The monthly gem5 Developers' Meeting will be held on October 10th at 4pm UTC.

If you would like to add anything to the agenda please create a comment below or reach out to Jason (@powerjg; [jason@lowepower.com](mailto:jason@lowepower.com)).

### Archive
Video recording: [YouTube](https://youtu.be/u8TvQMEnqPM)

**Comment by giactra** [comment]
Hi, I would like to discuss ongoing prefetcher fixes/additions:

https://github.com/gem5/gem5/pull/1441
https://github.com/gem5/gem5/pull/1454
https://github.com/gem5/gem5/pull/1436
https://github.com/gem5/gem5/pull/1461

And the following branch prediction discussion:

https://github.com/orgs/gem5/discussions/1341

**Comment by erin-le** [comment]
# October 10 gem5 Developers' Meeting Notes:

release date: moved from dec 4 to November 22nd, 1.5 month of dev time left

## 1st discussion: new prefetchers

- Authors couldn't make it, not worth having a discussion
- Giacomo: having these prefetcher PRs come in means that prefetchers aren't as good as we thought
- recommends coming up with microbenchmark tests for each. Maybe it could be a student assignment?
- people at Arm don't think the prefetchers are good

- Giacomo thinks reviewing these can be parallelized, Jason wants someone to look over all four

- One PR author didn't respond to requested software engineering changes

- Giacomo: if something is new, we can say we don't have bandwidth, but if someone is fixing something is broken, we should compromise

- Suggestion: Ping authors, discuss things with them
- "Shepherding": Someone acts as a point person to talk with authors. Giacomo will do it

## Issue: Underperforming state of art branch predictors

- Need to find a way to incentivize people to do impactful, but high effort work like this issue

## difficulty using extract-vmlinux to obtain ARM linux kernel

- when kernel unzipped, format not readable by bootloader

- Hoa: compile module w/out using extracted kernel
    - Jason doesn't think this is sustainable; if we can use the kernel, it simplifies the process. If we don't have the kernel, we have to do things separately for each kernel we support

- gem5 people will post an issue

##

- Discussion about various channels: Slack doesn't show history past a few months since we have free plan; no plans to upgrade due to cost

- GPU standard library

- separate function for adding a gpu workload to a (gpu) board? ie a function similar to set_kernel_disk_workload 

- PCI device: read - bogus data, write, panic
    - there is a workaround which warns instead of panics
- ACPI

- need documentation on where the "bandaid" solutions are

- Ruby static
